codex/rest/api.nim

```diff
@@ -218,27 +218,32 @@ proc initRestApi*(node: CodexNodeRef, conf: CodexConf): RestRouter =
   router.rawApi(
     MethodPost,
     "/api/codex/v1/upload") do (
+      chunk: Option[uint]
     ) -> RestApiResponse:
-      ## Upload a file in a streamming manner
+      ## Upload a file in a streamming manner,
+      ## split it into blocks of `chunk` bytes,
+      ## and save to the node's BlockStore
       ##
 
       trace "Handling file upload"
-      var bodyReader = request.getBodyReader()
+      let bodyReader = request.getBodyReader()
       if bodyReader.isErr():
-        return RestApiResponse.error(Http500)
+        return RestApiResponse.error(Http500, bodyReader.error)
 
-      # Attempt to handle `Expect` header
-      # some clients (curl), wait 1000ms
-      # before giving up
-      #
+      # Attempt to handle `Expect` header some clients (curl),
+      # wait 1000ms before giving up
       await request.handleExpect()
 
-      let
-        reader = bodyReader.get()
+      let bodyStream = AsyncStreamWrapper.new(reader = AsyncStreamReader(bodyReader.get))
 
       try:
-        without cid =? (
-          await node.store(AsyncStreamWrapper.new(reader = AsyncStreamReader(reader)))), error:
+        let runUpload =
+          if (chunkOpt =? chunk) and (blockSize =? chunkOpt) and (blockSize <= MaxBlockSize):
+            node.store(bodyStream, blockSize = blockSize.int)
```

Comment:
Why the double layered `=?`? Is `chunk` is already optional?

Reply:
yeah, I printed chunk variable contents with echo:
```
None[Result[system.uint, system.cstring]]      # no ?chunk parameter at all
------------------
Some(err(Non-decimal character encountered))   # ?chunk=-2
------------------
Some(ok(1000000000))           # ?chunk=1000000000
```

The type is `Opt[ Result[uint,cstring]]`, so external Opt is set to None when there is no "?chunk" in URL at all. And internal Result is used to return the parsing error message.